src/internal/_set.js

```diff
@@ -0,0 +1,23 @@
+var _extend = require('./_extend');
+var _has = require('./_has');
+var _isArray = require('./_isArray');
+var isEmpty = require('../isEmpty');
+var keys = require('../keys');
+
+module.exports = function _set(update, original) {
+  if (_isArray(original) || _isArray(update)) {
```

Comment:
Nit: I would prefer

```js
  if (_isArray(update) || _isArray(original))
```

just in order to keep a parallel with the parameters.